web/client/utils/GeoProcessingUtils.js

```diff
@@ -42,3 +45,30 @@ export const getCounter = (layers, groupName) => {
     const counter = allLayers.length ? Number(last(allLayers).name.match(/\d/)[0]) + 1 : 0;
     return counter;
 };
+
+/**
+ * it creates an id of a feature if not existing
+ * @param {object} feature list of layers to check
+ * @param {string} index the index of the feature
+ * @return {string} a predictable id
+ */
+export const createFeatureId = (feature, index) => {
+    return feature.id || `Feature #${index}`;
+};
+/**
+ * it creates an arc line into a multiple segments, only for LineString ft
+ * @param {object} feature list of layers to check
+ * @return {object} the transformed feature if geodesic
+ */
+export const transformLineToArc = (feature) => {
+    if (feature?.properties?.geodesic) {
```

Comment:
If it handles only linestrings, than do this check before transformation.

Reply:
what about adding a condition check for the geometry property? doing it external to the function is worse imo